test/libsolidity/SolidityOptimizer.cpp

```diff
@@ -485,7 +485,16 @@ BOOST_AUTO_TEST_CASE(constant_optimization_early_exit)
 	auto start = std::chrono::steady_clock::now();
 	compileBothVersions(sourceCode);
 	double duration = std::chrono::duration<double>(std::chrono::steady_clock::now() - start).count();
+	// Since this check takes close to 60 seconds on an ASan build,
+	// we disable it if we are compiling an ASan test binary. The
+	// DISABLE_CHECK_FOR_ASAN_BUILD macro is defined if
+	// -DSANITIZE=address is passed as a compile time cmake option.
+#ifndef DISABLE_CHECK_FOR_ASAN_BUILD
```

Comment:
I might have gone with just calling this ``ASAN_BUILD``... and we still want to run the test, even though it takes a long time? We could also move the ``#ifndef`` around the whole test to avoid having to wait for it for 60 seconds...

Reply:
You don't need to define own macro. There is a way to check if you are building with ASan: https://clang.llvm.org/docs/AddressSanitizer.html#conditional-compilation-with-has-feature-address-sanitizer.

Reply:
@bshastry noted that this is not cross-compiler. Does asan exist on non-llvm compilers?

Reply:
This will work for cross-compilation - the code will be instrumented for the target architecture (if that's a concern).

ASan is also supported with GCC. The same basic flags are used, but the runtime library is different I believe. In case of GCC the `__SANITIZE_ADDRESS__` macro is defined: https://gcc.gnu.org/onlinedocs/gcc-4.8.0/cpp/Common-Predefined-Macros.html.

Reply:
I changed it to use the compiler-internal check for now. It's fine for the test to fail on gcc.